src/prefect/client/base.py

```diff
@@ -154,27 +155,53 @@ class PrefectHttpxClient(httpx.AsyncClient):
 
     RETRY_MAX = 5
 
-    async def send(self, *args, **kwargs) -> Response:
-        retry_count = 0
-        response = PrefectResponse.from_httpx_response(
-            await super().send(*args, **kwargs)
-        )
+    async def _send_with_retry(
+        self,
+        request: Callable,
+        retry_codes: Set[int] = {},
+        retry_exceptions: Tuple[Exception, ...] = (),
+    ):
+        try_count = 0
+        response = None
+
         while (
-            response.status_code
-            in {status.HTTP_429_TOO_MANY_REQUESTS, status.HTTP_503_SERVICE_UNAVAILABLE}
-            and retry_count < self.RETRY_MAX
+            not response
+            or (response.status in retry_codes)
+            and try_count <= self.RETRY_MAX
         ):
-            retry_count += 1
+            try:
+                response = await request()
+            except retry_exceptions as exc:
+                if try_count >= self.RETRY_MAX:
+                    raise exc
```

Comment:
```suggestion
                    raise
```
including the exc adds a frame to the traceback